packages/format-currency/src/index.ts

```diff
@@ -47,11 +47,20 @@ export function createFormatter(): CurrencyFormatter {
 		code: string,
 		options: CurrencyObjectOptions
 	): Intl.NumberFormat {
-		return getCachedFormatter( {
-			locale: getLocaleToUse( options ),
+		const numberFormatOptions: Intl.NumberFormatOptions = {
+			style: 'currency',
 			currency: code,
-			noDecimals: isNoDecimals( number, options ),
-			signForPositive: options.signForPositive ?? false,
+			...( options.stripZeros &&
+				Number.isInteger( number ) && {
+					maximumFractionDigits: 0,
+					minimumFractionDigits: 0,
+				} ),
```

Comment:
I just saw my note below about the `trailingZeroDisplay` [option](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#trailingzerodisplay). That might be a better way to do this now that it seems to be well supported! If you'd prefer not to, could you retain the comment but maybe move it up here so that setting the digits manually can eventually be replaced?

Reply:
@sirbrillig the best chance right now, from my exploratikns, is that all of this code will migrate to i18n-calypso, then from there some generic parts to a separate package. So I think let's leave this chznge fof later, as we also set these values for the numberFofmat wrappers there. We'll eventually clean it all up. Just fof now  I'm more concerned about finding as many connections as possible.

Reply:
Of course. Added back^